frontend/static/js/index.js

```diff
@@ -24,18 +25,32 @@ const router = async () => {
         { path: "/", view: Dashboard },
         { path: "/posts", view: Posts },
         { path: "/posts/:id", view: PostView },
+        { path: "/posts/:id/category/:categoryId", view: PostViewCategories },
         { path: "/settings", view: Settings }
     ];
 
     // Test each route for potential match
     const potentialMatches = routes.map(route => {
+        
         return {
             route: route,
             result: location.pathname.match(pathToRegex(route.path))
+            
         };
     });
 
-    let match = potentialMatches.find(potentialMatch => potentialMatch.result !== null);
+
+    var match
+    potentialMatches.forEach(entry =>{
+
+         if(entry.result != null) {
+
+            match = (match === undefined || entry.route.path.length > match.route.path.length) ? entry : match
+         }
+              
+    })
```

Comment:
This one should be looked at again more closely.
In addition to an `=` at the beginning, a semicolon is also missing at the end too.

There are also unnecessary empty lines here, again that can be removed.